lib/ansible/modules/cloud/vmware/vmware_guest_register_operation.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, sky-joker
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+module: vmware_guest_register_operation
+short_description: VM inventory registration operation
+author:
+  - sky-joker (@sky-joker)
+version_added: '2.9'
+description:
+  - This module can register or unregister VMs to the inventory.
+requirements:
+  - python >= 2.6
+  - PyVmomi
+options:
+  datacenter:
+    description:
+    - Destination datacenter for the register/unregister operation.
+    - This parameter is case sensitive.
+    default: ha-datacenter
+  cluster_name:
+    description:
+      - Specify a cluster name to register VM.
+  folder:
+    description:
+    - Description folder, absolute path of the target folder.
+    - The folder should include the datacenter. ESX's datacenter is ha-datacenter.
+    - This parameter is case sensitive.
+    - 'Examples:'
+    - '   folder: /ha-datacenter/vm'
+    - '   folder: ha-datacenter/vm'
+    - '   folder: /datacenter1/vm'
+    - '   folder: datacenter1/vm'
+    - '   folder: /datacenter1/vm/folder1'
+    - '   folder: datacenter1/vm/folder1'
+  name:
+    description:
+    - Specify VM name to be registered in the inventory.
+    required: True
+  uuid:
+    description:
+    - UUID of the virtual machine to manage if known, this is VMware's unique identifier.
+    - If virtual machine does not exists, then this parameter is ignored.
+  esxi_hostname:
+    description:
+    - The ESXi hostname where the virtual machine will run.
+    - This parameter is case sensitive.
+  template:
+    description:
+    - Whether to register VM as a template.
+    default: False
+  path:
+    description:
+    - Specify the path of vmx file.
+    - 'Examples:'
+    - '    [datastore1] vm/vm.vmx'
+    - '    [datastore1] vm/vm.vmtx'
+  resource_pool:
```

Comment:
If you set both cluster_name and resource_pool, resource_pool seems to be ignored and the guest is created at the top level of the cluster.  If I leave cluster_name off, and just set resource_pool, the guest creates in the right place.

Reply:
This assumes that only one of the cluster or resource_pool is specified.
If the `and` conditions for the cluster and resource_pool are required, modify the code.